### PR TITLE
[BE] move general script out of .circleci/ into tools/

### PR DESCRIPTION
**Pull request by walterddr**
Second step in #60373.

### CODE REVIEW

File: tools/stats/upload_binary_size_to_scuba.py [hunk 1]
```diff
@@ -11,8 +11,10 @@
 
 import requests
 
+from typing import Any, Dict, Generator, List, Tuple
```
Comment (walterddr) [inline tools/stats/upload_binary_size_to_scuba.py:14]:
```suggestion
from typing import Any, Dict, Generator, List
```

File: tools/stats/upload_binary_size_to_scuba.py [hunk 2]
```diff
@@ -22,12 +24,11 @@ def get_size(file_dir):
...
-    pkg_type, py_ver, cu_ver, *_ = os.environ.get("BUILD_ENVIRONMENT", "").split() + [
-        None,
-        None,
-        None,
-    ]
+def build_message(size: int) -> Dict[str, Any]:
+    if "BUILD_ENVIRONMENT" in os.environ.keys():
+        pkg_type, py_ver, cu_ver, *_ = os.environ.get("BUILD_ENVIRONMENT", "").split()
+    else:
+        pkg_type, py_ver, cu_ver = "", "", ""
```
Comment (samestep) [inline tools/stats/upload_binary_size_to_scuba.py:31]: This doesn't seem to match up with what the original code meant; could you explain why this is ok?

**Review by samestep — COMMENTED** [review]
looks mostly good to me, except one change to the `build_message` function that I'm a bit concerned about

File: tools/stats/upload_binary_size_to_scuba.py (hunk 2 above)
Comment (walterddr) [reply tools/stats/upload_binary_size_to_scuba.py:31]:
```suggestion
    if "BUILD_ENVIRONMENT" in os.environ.keys():
        pkg_type, py_ver, cu_ver, *_ = os.environ.get("BUILD_ENVIRONMENT", "").split()
    else:
        pkg_type, py_ver, cu_ver = (None, None, None)
```

Comment (walterddr) [reply tools/stats/upload_binary_size_to_scuba.py:31]: hmm. you are right. I made this change b/c mypy complains concatenate string and None.

Comment (samestep) [reply tools/stats/upload_binary_size_to_scuba.py:31]: if you want to satisfy `mypy`'s pickiness then you could do this instead:
```suggestion
    from typing import Optional, Sequence
    split_env: Sequence[Optional[str]] = os.environ.get("BUILD_ENVIRONMENT", "").split()
    pkg_type, py_ver, cu_ver, *_ = list(split_env) + [None, None, None]
```

Comment (walterddr) [reply tools/stats/upload_binary_size_to_scuba.py:31]:
```suggestion
    build_env_split: List[Any] = os.environ.get("BUILD_ENVIRONMENT", "").split()
    pkg_type, py_ver, cu_ver, *_ = build_env_split + [None, None, None]
```

**Comment by samestep** [comment]
I notice that you seem to be modifying `third_party/benchmark`; is that intentional?

**Comment by walterddr** [comment]
> I notice that you seem to be modifying `third_party/benchmark`; is that intentional?

nope. not sure why it was included. fixing

**Review by samestep — APPROVED** [review]

**Pull request closed without merging**